core/src/com/unciv/ui/components/widgets/ZoomableScrollPane.kt

```diff
@@ -263,13 +260,16 @@ open class ZoomableScrollPane(
             onZoomStopListener?.invoke()
         }
 
-        override fun zoom(initialDistance: Float, distance: Float) {
-            if (lastInitialDistance != initialDistance) {
-                lastInitialDistance = initialDistance
-                lastScale = scaleX
-            }
-            val scale: Float = sqrt((distance / initialDistance).toDouble()).toFloat() * lastScale
-            zoom(scale)
+        override fun zoom(delta: Float, focus: Vector2) {
+            // correct map position to zoom in to the focus point
+            val dx = (stage.width / 2 - focus.x) * (delta - 1) / scaleX
+            val dy = (stage.height / 2 - focus.y) * (delta - 1) / scaleY
```

Comment:
This needs a lot more explanation, in comments.
scaleX and scaleY should always be the same so this is IMO needlessly confusing.

"stage and focus and in **stage pixesl**. Since the entire map is zoomed, however, the actual on-screen pixels are in stage pixels * scale".

"We take the diff between the focus point and the stage center point - that's the difference in stage pixels."
"When scrolling, it would be a different number of pixels - to get this need to convert *current zoom pixels* to *stage pixels* to *new zoom pixels".
"Stage to zoom pixels are 1:scale, so current zoom pixels to new zoom pixels are oldScale:newScale"
"Since newScale = oldScale * delta, oldScale:newScale = oldScale:oldScale*delta

And from here tbh I lost you

Reply:
I added comments in the code, let me know if it's easier to understand now. Key point is that translation consists of two parts: one cause by moving the focal point and another one caused by scale change, because we want to direct zooming towards the focal point. The latter is what was calculated here. I moved that calculation for better clarity.